partcad/src/partcad/test/cam.py

```
@@ -0,0 +1,117 @@
+        if not is_part and not is_assembly:
+            self.debug(shape, "Not applicable")
+            return True
+
+        if not shape.is_manufacturable and not "force_manufacturing" in test_ctx:
+            self.debug(shape, "Not supposed to be manufacturable")
+            return True
+
+        if isinstance(shape, Part):
+            return await self.test_part(tests_to_run, ctx, shape, test_ctx)
+        elif isinstance(shape, Assembly):
+            return await self.test_assembly(tests_to_run, ctx, shape, test_ctx)
```

Ruff (0.8.2)

Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)